### PR TITLE
fix: gas estimation for adding liquidity

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: david <david@umaproject.org>

We have to estimate based on gas limit and the max fee per gas that could be used, otherwise the transaction reverts

### CODE REVIEW

File: src/utils/chains.ts [hunk 1]
```diff
@@ -40,5 +40,6 @@ export function isSupportedChainId(chainId: number): chainId is ChainId {
 export async function getGasPrice(
   provider: ethers.providers.JsonRpcProvider
 ): Promise<ethers.BigNumber> {
-  return provider.getGasPrice();
+  const fees = await provider.getFeeData();
+  return fees.maxFeePerGas || fees.gasPrice || (await provider.getGasPrice());
 }
```
Comment (Tulun) [inline src/utils/chains.ts:45]: Could this await call fail?

**Review by Tulun — APPROVED** [review]
Just a quick comment, otherwise LGTM

File: src/utils/chains.ts (hunk 1 above)
Comment (daywiss) [reply src/utils/chains.ts:45]: hm probably

Comment (daywiss) [reply src/utils/chains.ts:45]: added a catch for the caller

**Pull request merged**